### PR TITLE
[ROCm] Remove PYTORCH_MIOPEN_SUGGEST_NHWC flag

### LINKED ISSUE #64427: Remove PYTORCH_MIOPEN_SUGGEST_NHWC flag in ROCm
PYTORCH_MIOPEN_SUGGEST_NHWC is an environment flag to manually recommend NHWC. The flag is added during PR: https://github.com/pytorch/pytorch/pull/63617

Once MIOpen does officially supports NHWC, remove this flag, and switch on ROCm version. 

cc @jeffdaily @sunway513 @jithunnair-amd @ROCmSupport

### PR DESCRIPTION
**Pull request by jeffdaily**
Fixes #64427.  MIOpen supports ChannelsLast.  No longer need to opt-in with env var.

cc @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by malfet — APPROVED** [review]

File: test/nn/test_convolution.py [hunk 1]
```diff
@@ -2183,7 +2183,7 @@ def helper(n, c, h, w, out_channels, kernel_size, groups):
             out = conv(input)
             ref_out = ref_conv(ref_input)
 
-            grad = torch.randint_like(out, -3, 3)
+            grad = torch.randint_like(out, -3, 3, dtype=torch.int64).to(dtype=dtype)
```
Comment (malfet) [inline test/nn/test_convolution.py:2186]: It would be good to add a short comment why this change is necessary.

File: test/nn/test_convolution.py [hunk 2]
```diff
@@ -2160,18 +2159,19 @@ def helper(n, c, h, w, out_channels, kernel_size, dilation, groups, input_format
...
     @skipCUDAIfCudnnVersionLessThan(7603)
+    # randint and randint_like with dtype=torch.cfloat raises
+    # RuntimeError: check_random_bounds handles only integral, floating-point and boolean types
     @dtypes(torch.half, torch.float, torch.cfloat)
     def test_conv_cudnn_nhwc(self, device, dtype):
         def helper(n, c, h, w, out_channels, kernel_size, groups):
-            input = torch.randint(-3, 3, (n, c, h, w), dtype=dtype, device=device)\
-                .to(memory_format=torch.channels_last)
+            input = torch.randint(-3, 3, (n, c, h, w), device=device)\
+                .to(memory_format=torch.channels_last, dtype=dtype)
```
Comment (malfet) [inline test/nn/test_convolution.py:2169]: This looks semantically equivalent to the previous statement, but adding a line or two explaining why this is necessary would be nice

Comment (jeffdaily) [reply test/nn/test_convolution.py:2169]: lines 2163 and 2164 are the comment explaining the change
